src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.ts

```diff
@@ -61,10 +61,12 @@ export class ExamChecklistComponent implements OnInit {
         // calculate mandatory points and optional points
         if (this.pointsExercisesEqual) {
             this.exam.exerciseGroups!.forEach((exerciseGroup) => {
-                if (exerciseGroup.isMandatory) {
-                    sumPointsExerciseGroupsMandatory += exerciseGroup!.exercises![0]!.maxPoints!;
-                } else {
-                    sumPointsExerciseGroupsOptional += exerciseGroup!.exercises![0]!.maxPoints!;
+                if (exerciseGroup!.exercises && exerciseGroup.exercises.length !== 0) {
```

Comment:
You probably don't need the `!` after `exerciseGroup` here and below, but I guess it doesn't hurt either.